### PR TITLE
Subf2m provider improvements

### PR DESCRIPTION
**Pull request by silentcommitter**
Supports more languages.

Changes the movie and tv show title matching from searching for the title's complete inclusion in the html text which would skip correct titles occasionally, as well as pick wrong titles. Instead the title matching now uses regex to more accurately match correct season (tvshow) or year (movie). To match the title difflib is used to calculate string similarity and the most similar string is picked as the desired title.

### CODE REVIEW

**Comment by morpheus65535** [comment]
@vitiko98 looks good to me, are you good with that?

**Comment by vitiko98** [comment]
Thanks for this PR!

All tests were passed. However, could you provide some examples for your new match logic? Just to make sure all it's working perfectly. You can do it adding tests to `tests/subliminal_patch/test_subf2m.py` or just giving me series/movie titles or links so I can add it myself.

**Comment by silentcommitter** [comment]
If you could add a test for "house of the dragon season 1" as well as Pirates of the Caribbean Dead Man's Chest, that would demonstrate the improvements i believe. House of the dragon would previously be matched to "[On The Next House Of The Dragon - First Season](https://subf2m.co/subtitles/on-the-next-house-of-the-dragon-first-season)" and Dead Man's Chest as it's titled [Pirates of the Caribbean 2: Dead Man's Chest (2006)](https://subf2m.co/subtitles/pirates-of-the-caribbean-2-dead-mans-chest) at subf2m (note the "2")

File: libs/subliminal_patch/providers/subf2m.py [hunk 1]
```diff
@@ -82,12 +85,36 @@ def id(self):
...
+    "german": "ger",
+    "greek": "gre",
+    "hungarian": "hun",
+    "icelandic": "ice",
+    "japanese": "jpn",
+    "macedonian": "mac",
+    "malay": "may",
+    "polish": "pol",
+    "romanian": "rum",
+    "russian": "rus",
```
Comment (vBm) [inline libs/subliminal_patch/providers/subf2m.py:107]: You can add `Serbian` as well being that it is supported on `subf2m`
`"serbian": "srp",`

**Review by vitiko98 — APPROVED** [review]

**Comment by vitiko98** [comment]
Everything great.

**Pull request merged**